src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -123,29 +122,7 @@ internal partial class Binder
 
         private bool IsCheckedConversion(TypeSymbol source, TypeSymbol target)
         {
-            Debug.Assert((object)target != null);
-
-            if ((object)source == null || !CheckOverflowAtRuntime)
-            {
-                return false;
-            }
-
-            if (source.IsDynamic())
-            {
-                return true;
-            }
-
-            SpecialType sourceST = source.StrippedType().EnumUnderlyingType().SpecialType;
-            SpecialType targetST = target.StrippedType().EnumUnderlyingType().SpecialType;
-
-            // integral to double or float is never checked, but float/double to integral 
-            // may be checked.
-            bool sourceIsNumeric = SpecialType.System_Char <= sourceST && sourceST <= SpecialType.System_Double;
-            bool targetIsNumeric = SpecialType.System_Char <= targetST && targetST <= SpecialType.System_UInt64;
-
-            return
-                sourceIsNumeric && (targetIsNumeric || target.IsPointerType()) ||
-                targetIsNumeric && source.IsPointerType();
```

Comment:
It looks like `NeedsCheckedConversion` handles more case than this method used to. The new method checks `sourceIsNumeric && targetIsNumeric && (!compareNumericTypes || s_needsChecked[sourceST - SpecialType.System_Char, targetST - SpecialType.System_Char]) `.

Does that mean we fixed another gap by this refactoring? If that gap exists and can be identified, maybe it should be tested.